src/dropdown/Dropdown.js

```diff
@@ -199,9 +194,25 @@ export default class Dropdown extends React.Component {
   }) => {
 
     const Option = optionComponent;
-    const propertyName = this.props.groupByKey;
+    const { groupByKey } = this.props;
+    const groupedOptions = options.reduce((acc, o, i) => {
+      const shouldCreateNewGroup = i === 0 || o[groupByKey] !== options[i - 1][groupByKey];
```

Comment:
this logic works because you are relying on the fact that `options` area already sorted, right? if yes, can you add a very short comment?